### ISSUE TITLE
Module#parents has been renamed to `module_parents`

### ISSUE THREAD

**Issue by cilim** [makandra/active_type#121]
Hi there 👋 

Love your gem! Keep up the good work 🔧 

`activesupport-6.0.3.2` has added a deprecation warning in `Introspection` module:

```ruby
def parents
  ActiveSupport::Deprecation.warn(<<-MSG.squish)
    `Module#parents` has been renamed to `module_parents`.
    `parents` is deprecated and will be removed in Rails 6.1.
  MSG
  module_parents
end
```

`active_type (1.4.0)` is using `.parents` in `active_type/record_extension/inheritance.rb`:
```ruby
extended_record_base_class.parents
```

which results in many deprecation warning when running the specs:

```
DEPRECATION WARNING: Module#parents has been renamed to module_parents. parents is deprecated and will be removed in Rails 6.1.
```

Can you please fix this and release a new version for the gem? Thanks!

**Comment by triskweline** [comment]
Hi @cilim! Thanks for letting us know about this API change. Can you prepare a PR that fixes this?

**Comment by cilim** [comment]
Gladly @triskweline!
Just wanted to confirm with you before I start that this might be up for bumping the major version, because `.module_parents` has been introduced on [18th January 2019](https://github.com/rails/rails/blob/v6.0.2.2/activesupport/CHANGELOG.md#rails-600beta1-january-18-2019) and doesn't exist in activesupport versions that are `> 6`.

That means we'd also have to change the `active_type.gemspec` `'activerecord', '>= 6.0'` to assure the method exists.

Let me know what you think and if I should move forward with the PR.

**Comment by triskweline** [comment]
We currently support Rails 3.2+ and cannot break compatibility. This change would need to have a version switch, or check if the `Module` instance responds to `module_parents`.

We have a separate Gemfile for Rails 6 tests (they run in CI after pushing). It's OK to bump the Rails 6 version in that Gemfile, since we only support the highest minor/patch release for each major version.

**Comment by cilim** [comment]
> This change would need to have a version switch, or check if the Module instance responds to module_parents.

Since it's really easy to use `respond_to?` I'd go with this option and just bump the minor version if that's ok with you.

**Comment by triskweline** [comment]
Yes, using `respond_to?` would work great for this. Thanks for your help!

**Comment by cilim** [comment]
PR https://github.com/makandra/active_type/pull/122 ready!

**Comment by triskweline** [comment]
I released this change as 1.4.1. Thanks for the PR!
